pkg/svc/zserver.go

```diff
@@ -2,51 +2,142 @@ package svc
 
 import (
 	"context"
+	"flag"
+	"time"
 
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/my-app/pkg/pb"
+	rpb "github.com/my-app/pkg/resp_pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~ A BRIEF DEVELOPMENT GUIDE ~~~~~~~~~~~~~~~~~~~~~~~~~
-//
-// TODO: Extend the MyApp service by defining new RPCs and
-// and message types in the pb/service.proto file. These RPCs and messages
-// compose the API for your service. After modifying the proto schema in
-// pb/service.proto, call "make protobuf" to regenerate the protobuf files.
-//
-// TODO: Create an implementation of the MyApp server
-// interface. This interface is generated by the protobuf compiler and exists
-// inside the pb/service.pb.go file. The "server" struct already provides an
-// implementation of MyApp server interface service, but only
-// for the GetVersion function. You will need to implement any new RPCs you
-// add to your protobuf schema.
-//
-// TODO: Update the GetVersion function when newer versions of your service
-// become available. Feel free to change GetVersion to better-suit how your
-// versioning system, or get rid of it entirely. GetVersion helps make up a
-// simple "starter" example that allows an end-to-end example. It is not
-// required.
-//
-// TODO: Oh yeah, delete this guide when you no longer need it.
-//
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~ FAREWELL AND GOOD LUCK ~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
 const (
 	// version is the current version of the service
 	version = "0.0.1"
 )
 
+var (
+	serverAddr = flag.String("addr", "localhost:9099", "The server address in the format of host:port")
+)
+
 // Default implementation of the MyApp server interface
-type server struct{}
+type server struct {
+	pb.UnimplementedMyAppServer
+	Description string
+	Timestamp   time.Time
+	Requests    int64
+}
+
+func GRPCConnect() (*grpc.ClientConn, error) {
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithInsecure())
+	conn, err := grpc.Dial(*serverAddr, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
 
 // GetVersion returns the current version of the service
-func (server) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
-	return &pb.VersionResponse{Version: version}, nil
+func (*server) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
+	conn, err := GRPCConnect()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := rpb.NewMyResponderClient(conn)
+	resp, err := client.GetVersion(context.Background(), &empty.Empty{})
+	if err != nil {
+		return nil, err
+	}
+	return &pb.VersionResponse{Version: resp.Version}, nil
+}
```

Comment:
Not sure that we need this function, because we did not have such a requirement in our task. Such code should not be included and released.

Reply:
Sorry for that, i forgot about this function. It's useless and i will delete it

Reply:
Fixed